src/libs/antares/study/parts/common/cluster_list.cpp

```diff
@@ -463,9 +463,6 @@ bool ClusterList<ClusterT>::remove(const Data::ClusterName& id)
     // Invalidating the parent area
     c->parentArea->invalidate();
 
```

Comment:
This method remove(id) is modified (si its behavior is modified) , but it is used elsewhere in the code, for example in : 
**src\ui\simulator\toolbox\input\thermal-cluster.cpp**
So this change could have nasty side effect

Reply:
There are two calls for `remove`. One is in `removeDisabledClusters`, the other is where you mentioned.

The only side-effect is that some memory could still be accessible after closing the program (memory leak). So here is my suggestion :
- Do not free allocated memory in method `remove`
- Free everything in the destructor `~ClusterList`.

Reply:
There are two maps for thermal clusters
* `cluster`, defined in `common/cluster_list.h`
* mapping, defined in `renewable/cluster_list.h`
These maps contain copies of the same shared pointers. When the user adds a thermal cluster, the two following lines are executed
```cpp
auto added = pArea->thermal.list.add(cluster);
pArea->thermal.list.mapping[cluster->id()] = added;
```
In other words, both `mapping` and `cluster` contain the same pointer. There are **2** references to the pointed data.

When the user deletes a thermal cluster, the following line is executed
```cpp
pArea->thermal.list.remove(toDelete->id())
```
The remove method is `ThermalClusterList::remove()` :
```cpp
bool ThermalClusterList::remove(const ClusterName& id)
{
    auto i = cluster.find(id);
    if (i == cluster.end())
        return false;

    // Getting the pointer on the cluster
    SharedPtr c = i->second;

    // Removing it from the list
    cluster.erase(i);
    // Invalidating the parent area
    c->parentArea->invalidate();

    // Remove all cluster coupling
    if (not c->coupling.empty())
    {
        auto end = c->coupling.end();
        for (auto j = c->coupling.begin(); j != end; ++j)
        {
            auto* link = *j;
            link->parentArea->invalidate();
            link->coupling.erase(c.get());
        }
    }

    // Rebuilding the index
    rebuildIndex();
    return true;
}
```
Only **1** reference remains, from the shared pointer in `mapping`. However, when the user closes the GUI, the destructor of this structure is called, which calls the destructors of the `shared_ptr`, so there is **0** reference left. Therefore, the `ThermalCluster` pointer is automatically deallocated.